konnectintech/Konectin-website-_backend
# Fix internship application endpoint

# Pull Request: Implement Internship Application Endpoint

## Description
Updated the internship application endpoint that allows users to submit their applications with basic details and resume files.

## Changes Made
1. Updated internship application controller with file upload handling and relevant required fields (some fields not present on the design have been added because they are required in the internship schema)

## Testing
- Added validation for enum values (ageRange, preferedField, internshipType)
- Tested file upload functionality with size and type restrictions
- Verified proper error handling for invalid inputs

## API Documentation

### POST /internshipApplication
Submit a new internship application

**Request Body:**
```json
{
  "fullName": "string",
  "email": "string",
  "countryCode": "string",
  "phoneNumber": "string",
  "country": "string",
  "ageRange": "string (enum)",
  "portfolio": "string (URL)",
  "preferedField": "string (enum)",
  "internshipType": "string (enum)",
  "resumes": "file(s)"
}
```

**Valid Enum Values:**
- ageRange: ["18yrs-24yrs", "25yrs-34yrs", ...]
- preferedField: ["Software Development(Frontend)", "Software Development (Backend)", ...]
- internshipType: ["Paid", "Unpaid", "Both"]

## Testing Instructions
1. Use Postman to test the endpoint
2. Set request type to POST
3. Use form-data to include file uploads
4. Ensure all required fields are included with valid enum values


## Screenshots
![image](https://github.com/user-attachments/assets/451fdbee-3ef8-4f12-a79a-2c45b6c9f6c0)
![image](https://github.com/user-attachments/assets/12cb16a6-27bc-4b6f-8679-f656b203b7d3)


## Additional Notes
The PR implements the endpoint properly and securely.

@Michaelajayi150
